subsys/mgmt/ec_host_cmd/backends/ec_host_cmd_backend_spi_stm32.c

```diff
@@ -745,6 +782,74 @@ struct ec_host_cmd_backend *ec_host_cmd_backend_get_spi(struct gpio_dt_spec *cs)
 	return &ec_host_cmd_spi;
 }
 
+#ifdef CONFIG_PM_DEVICE
+static int ec_host_cmd_spi_stm32_pm_action(const struct device *dev,
+					   enum pm_device_action action)
+{
+	const struct ec_host_cmd_backend *backend = (struct ec_host_cmd_backend *)dev->data;
+	struct ec_host_cmd_spi_ctx *hc_spi = (struct ec_host_cmd_spi_ctx *)backend->ctx;
+	const struct ec_host_cmd_spi_cfg *cfg = hc_spi->spi_config;
+	int err;
+
+	switch (action) {
+	case PM_DEVICE_ACTION_RESUME:
+		/* Set pins to active state */
+		err = pinctrl_apply_state(cfg->pcfg, PINCTRL_STATE_DEFAULT);
+		if (err < 0) {
+			return err;
+		}
+
+		/* Enable device clock */
+		err = clock_control_on(DEVICE_DT_GET(STM32_CLOCK_CONTROL_NODE),
+				       (clock_control_subsys_t)&cfg->pclken[0]);
+		if (err < 0) {
+			return err;
+		}
+		/* Enable CS interrupts. */
+		gpio_pin_interrupt_configure_dt(&hc_spi->cs, GPIO_INT_EDGE_BOTH);
+
+		break;
+	case PM_DEVICE_ACTION_SUSPEND:
+#ifdef SPI_SR_BSY
+		/* Wait 10ms for the end of transaction to prevent corruption of the last
```

Comment:
`WAIT_FOR` timeout arg unit is microseconds.

Reply:
oh good catch there, also now that I noticed:
- the last argument should probably be a longer sleep, since we are waiting up to 10ms, maybe `k_msleep(1)`?
- you can check the return value on exit, should be false if it timesout, then you can print an error

Reply:
Not sure about the fact of sleeping in PM_DEVICE_ACTION_SUSPEND. Typically, in CONFIG_PM_DEVICE, PM framework will cause the device suspension before going to stop mode. Calling a `k_msleep` will change its scheduling and possibly have a consequence on the current suspension. And we might end up in stopping the current suspension, before calling it again few ms later.

I don't say that it won't work, but at least the fact that all will happen seamlessly is not obvious and would require a comment.

Reply:
I see, then maybe just timeout handling and error message would do

Reply:
Not sure about the log message. 

Logging will get the whole system exit the sleep procedure, and depending on the case (PM, DEVICE, RUNTIME ?) trigger a chain of event (enable gpio clocks, configure uart pins, disable gpio clocks, enable uart clock, transmit chars, ...enable gpio clocks, disable uart pins, disable gpio clocks) and then start the system procedure once again, including this SPI subsystem.

So, should it be logged or not depends on the hypothesis made on logging. But in any case, it won't be the same behavior w/ and w/o logging.  I think it is worth some comments about general behavior expectations.

Note that I don't have an example of driver to give that should be taken as example, I just warn that things in this area are not obvious.

Reply:
Good point. Looking at conditions when the HC backend is being suspended, I think it is not necessary to print the message and potentially "stop" the suspend process of the system.